trinity/plugins/builtin/light_peer_chain_bridge/light_peer_chain_bridge.py

```diff
@@ -0,0 +1,253 @@
+    def expected_response_type() -> Type[BytesResponse]:
+        return BytesResponse
+
+
+class LightPeerChainEventBusResponder:
+    """
+    The ``LightPeerChainEventBusResponder`` listens for certain events on the eventbus and
+    delegates them to the ``LightPeerChain`` to get answers. It then propagates responses
+    back to the caller.
+    """
+
+    def __init__(self, chain: BaseLightPeerChain, event_bus: Endpoint) -> None:
+        self.chain = chain
+        self.event_bus = event_bus
+        asyncio.ensure_future(self.answer_get_blockheader_by_hash_requests())
+        asyncio.ensure_future(self.answer_get_blockbody_by_hash_requests())
+        asyncio.ensure_future(self.answer_get_receipts_by_hash_requests())
+        asyncio.ensure_future(self.answer_get_account_requests())
+        asyncio.ensure_future(self.answer_get_contract_code_requests())
+
+    async def answer_get_blockheader_by_hash_requests(self) -> None:
+        async for event in self.event_bus.stream(GetBlockHeaderByHashRequest):
+
+            val, error = await self._pass_or_raise(
+                self.chain.get_block_header_by_hash(event.block_hash)
+            )
+
+            self.event_bus.broadcast(
+                BlockHeaderResponse(val, error),
+                event.broadcast_config()
+            )
+
+    async def answer_get_blockbody_by_hash_requests(self) -> None:
+        async for event in self.event_bus.stream(GetBlockBodyByHashRequest):
+
+            val, error = await self._pass_or_raise(
+                self.chain.get_block_body_by_hash(event.block_hash)
+            )
+
+            self.event_bus.broadcast(
+                BlockBodyResponse(val, error),
+                event.broadcast_config()
+            )
+
+    async def answer_get_receipts_by_hash_requests(self) -> None:
+        async for event in self.event_bus.stream(GetReceiptsRequest):
+
+            val, error = await self._pass_or_raise(self.chain.get_receipts(event.block_hash))
+
+            self.event_bus.broadcast(
+                ReceiptsResponse(val, error),
+                event.broadcast_config()
+            )
+
+    async def answer_get_account_requests(self) -> None:
+        async for event in self.event_bus.stream(GetAccountRequest):
+
+            val, error = await self._pass_or_raise(
+                self.chain.get_account(event.block_hash, event.address)
+            )
+
+            self.event_bus.broadcast(
+                AccountResponse(val, error),
+                event.broadcast_config()
+            )
+
+    async def answer_get_contract_code_requests(self) -> None:
+
+        async for event in self.event_bus.stream(GetContractCodeRequest):
+
+            val, error = await self._pass_or_raise(
+                self.chain.get_contract_code(event.block_hash, event.address)
+            )
+
+            self.event_bus.broadcast(
+                BytesResponse(val, error),
+                event.broadcast_config()
+            )
```

Comment:
I tried that but I couldn't get it to work without giving up type safety. I think it boils down to the fact that:

a.) type inference still needs to improve
b.) specifying bounds is still rather limited (e.g. can't express `typeVar("TResponse", bound=BaseLightPeerChainResponse[TWhatever])`)

However, instead of saying:

```python
self.event_bus.broadcast(
    BytesResponse(val, error),
    event.broadcast_config()
)
```

We can say

```python
 self.event_bus.broadcast(
     event.expected_response_type()(val, error),
     event.broadcast_config()
)
```

So, at least we get rid of manually instantiating a response type.